components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1013,6 +1013,16 @@ bool ConversationDriver::HasPendingConversationEntry() {
   return !pending_conversation_entry_.is_null();
 }
 
+void ConversationDriver::StopGeneration() {
+  is_request_in_progress_ = false;
+  engine_->ClearAllQueries();
```

Comment:
nit: there _could_ be non-conversation queries in progress, such as suggested questions, or text rewriting, etc. It's probably a rare edge case which isn't important for this PR, but we could hold on to the API Request Helper Ticket for the current request so that we can cancel the specific conversation request only. A higher effort and not sure worth the reward.

Reply:
yes, i do have a todo here for [that](https://github.com/brave/brave-core/blob/master/components/ai_chat/core/browser/engine/remote_completion_client.cc#L207-L208)